source/hpack.c

```diff
@@ -502,26 +502,29 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
         return AWS_OP_ERR;
     }
 
-    /* Copy as much the above block as possible */
-    size_t above_block_size = context->dynamic_table.max_elements - context->dynamic_table.index_0;
-    if (above_block_size > new_max_elements) {
-        above_block_size = new_max_elements;
-    }
-    memcpy(
-        new_buffer,
-        context->dynamic_table.buffer + context->dynamic_table.index_0,
-        above_block_size * sizeof(struct aws_http_header));
-
-    /* Copy as much of below block as possible */
-    const size_t free_blocks_available = new_max_elements - above_block_size;
-    const size_t old_blocks_to_copy = context->dynamic_table.max_elements - above_block_size;
-    const size_t below_block_size =
-        free_blocks_available > old_blocks_to_copy ? old_blocks_to_copy : free_blocks_available;
-    if (below_block_size) {
+    /* Don't bother copying data if old buffer was of size 0 */
+    if (AWS_UNLIKELY(context->dynamic_table.max_elements > 0)) {
```

Comment:
Is this unlikely, or likely? Is the dynamic table THAT unlikely?

Reply:
Oh fuck I got that backwards